### PR TITLE
Fix exceptions during steps leaving pending-* files

### PR DESCRIPTION
**Pull request by YPares**
This should close https://github.com/tweag/funflow/issues/132

### CODE REVIEW

File: funflow/TestFunflow.hs [hunk 1]
```diff
@@ -32,22 +35,33 @@ flow2caught = retry 100 0 flow2
...
+      memHook <- createMemoryCoordinator
+      res <- runSimpleFlow MemoryCoordinator memHook store flow2 ()
+      print res
+      res' <- runSimpleFlow MemoryCoordinator memHook store flow2caught ()
+      print res'
+      putStrLn $ showFlow myFlow
+      putStrLn $ showFlow flow2
+      res1 <- runSimpleFlow MemoryCoordinator memHook store flow3 [1..10]
+      print res1
+  --  main = redisTest
```
Comment (aherrmann) [inline funflow/TestFunflow.hs]: Should that comment be removed?

Comment (YPares) [reply funflow/TestFunflow.hs]: Possibly, but I didn't edit it as part of this PR

**Comment by YPares** [comment]
@aherrmann I noticed something: for our project I have a `wrap'` task that takes 12sec to complete.

When Ctrl+Cing it right at the beginning, it takes 5secs before it actually stops. If I Ctrl+C **twice**, then it stops immediately, but that makes the exception handler cleanup part stop before it removed all the pending files. Is that to be expected?

**Comment by aherrmann** [comment]
IIRC the default signal handler is one-shot and a second Ctrl-C will just kill the process. You can avoid this by setting up a custom signal handler.

**Comment by regnat** [comment]
[Indeed](
https://stackoverflow.com/questions/2349233/catching-control-c-exception-in-ghc-haskell/7941166#7941166
)

**Pull request merged**